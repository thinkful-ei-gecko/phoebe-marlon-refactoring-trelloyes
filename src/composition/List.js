import React from 'react'
import './List.css';
import Card from './Card.js';

function List(props) {
  return (
      <section className="List">
        <header className="List-header">
          <h2>{props.header}</h2>
        </header>
          <div className="List-cards">
          {props.cardIds.map(obj => <Card title={obj.title} content={obj.content} key={obj.id} /> )}
        </div>
      </section>
  );
}


export default List;
