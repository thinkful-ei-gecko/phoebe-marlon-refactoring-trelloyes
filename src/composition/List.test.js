import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './list';

describe('<List />', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List header='okay' key='1' cardIds={[]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List header='Testlist' key='5' cardIds={['a','b']} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
})