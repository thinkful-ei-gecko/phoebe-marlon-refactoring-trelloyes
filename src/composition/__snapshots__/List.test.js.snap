// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      okay
    </h2>
  </header>
  <div
    className="List-cards"
  />
</section>
`;
